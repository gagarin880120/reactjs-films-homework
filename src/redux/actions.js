function searchResultsAction(results) {
  return {
      type: 'SEARCH',
      results: results
  };
}

function genresAction(genres) {
  return {
      type: 'GENRES',
      genres: genres
  };
}

export { searchResultsAction, genresAction };
