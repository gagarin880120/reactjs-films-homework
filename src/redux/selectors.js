import getGenresString from '../helpers/helpers';

const searchResultsSelector = (state) => state.searchResults.map((v) => ({
  ...v,
  genres: getGenresString(v.genre_ids, state.genres),
}));

const modalSelector = (state) => state.isModalOpen;

const trailerSelector = (state) => state.trailerURL;

export { searchResultsSelector, modalSelector, trailerSelector };
