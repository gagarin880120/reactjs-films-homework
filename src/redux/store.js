import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

const initialState = {
  searchResults: [],
  genres: [],
  isModalOpen: false,
  trailerURL: '',
  isTrailerLoaded: false,
};

export function reducer(state = initialState, action) {
  switch (action.type) {
    case 'SEARCH':
      return { ...state, searchResults: action.results };
    case 'GENRES':
      return { ...state, genres: action.genres };
    case 'MODAL':
      return { ...state, isModalOpen: action.isModalOpen };
    case 'TRAILER':
      return { ...state, trailerURL: action.trailerURL };
    case 'IS_TRAILER_LOADED':
      return { ...state, isTrailerLoaded: action.isTrailerLoaded };
    default:
      return state;
  }
}

export const store = createStore(reducer, applyMiddleware(thunk));
