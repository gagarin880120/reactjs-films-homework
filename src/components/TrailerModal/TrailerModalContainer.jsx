import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import TrailerModal from './TrailerModal';
import { modalSelector, trailerSelector } from '../../redux/selectors';
import { setModal } from '../../redux/actions';

export function TrailerModalContainer({ isModalOpen, closeModal, trailerURL }) {
  return (
    <TrailerModal
      isModalOpen={isModalOpen}
      closeModal={closeModal}
      trailerURL={trailerURL}
    />
  );
}

export const mapStateToProps = (state) => ({
  isModalOpen: modalSelector(state),
  trailerURL: trailerSelector(state),
});

export const mapDispatchToProps = (dispatch) => ({
  closeModal() {
    dispatch(setModal(false));
  },
});

TrailerModalContainer.propTypes = {
  isModalOpen: PropTypes.bool,
  closeModal: PropTypes.func,
  trailerURL: PropTypes.string,
};

TrailerModalContainer.defaultProps = {
  isModalOpen: false,
  closeModal: null,
  trailerURL: '',
};

export default connect(mapStateToProps, mapDispatchToProps)(TrailerModalContainer);
