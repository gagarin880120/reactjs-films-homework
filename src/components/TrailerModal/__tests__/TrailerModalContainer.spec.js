import ShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';
import { TrailerModalContainer, mapStateToProps, mapDispatchToProps } from '../TrailerModalContainer';
import { setModal } from '../../../redux/actions';

jest.mock('../../../redux/actions',() => ({
  setModal: jest.fn().mockReturnValue(false),
}))

describe('TrailerModalContainer', () => {
  describe('component', () => {
    test('should render without crashing', () => {
      const renderer = new ShallowRenderer();
      renderer.render(<TrailerModalContainer />);
      const result = renderer.getRenderOutput();
      expect(result).toMatchSnapshot();
    });
  })

  describe('mapDispatchToProps', () => {
    afterEach(() => {
      jest.clearAllMocks();
    })

    test('closeModal should dispatch setModal action', () => {
      const dispatch = jest.fn();
      const { closeModal } = mapDispatchToProps(dispatch);

      closeModal();

      expect(setModal).toHaveBeenCalled();
      expect(dispatch).toHaveBeenCalledWith(false);
    })
  })

  describe('mapStateToProps', () => {
    test('should return the right value', () => {
      const initialState = {
        isModalOpen: true,
        trailerURL: 'url'
      };
      expect(mapStateToProps(initialState).isModalOpen).toEqual(true);
    });
  });
})
