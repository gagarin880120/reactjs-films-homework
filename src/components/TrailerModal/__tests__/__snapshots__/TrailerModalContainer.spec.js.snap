// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrailerModalContainer component should render without crashing 1`] = `
<TrailerModal
  closeModal={[Function]}
  isModalOpen={false}
  trailerURL=""
/>
`;
